### PR TITLE
security: Windows update to 2024-12b

### PR DESCRIPTION
**Pull request by timmy-wright**
**What type of PR is this?**

/kind security

**What this PR does / why we need it**:

Windows 2024.12b patch release.

Due to an openssh permissions problem, we have to install the windows server 2019 patch as a patch onto the 2024-11b build. Hopefully this will be fixed in the next patch tuesday release for windows server 2019.

Because the VM extension is still getting applied in our test VHD build pipeline, I expect the VHDs in this PR to fail to build. So I've created a separate PR which was targeted to master and skipped the extension test - that will also be closer to our production build. The build pipeline for that PR is here. It initially failed on server 2019 for disk space issues as well, so I've contacted Evan Baker and removed a CNI/CNS container. That got us under the 30GB limit.

https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=110044034&view=results

**Which issue(s) this PR fixes**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 [hunk 1]
```diff
@@ -60,16 +60,17 @@ $global:defenderUpdateInfoUrl = "https://go.microsoft.com/fwlink/?linkid=870379&
...
 
         $global:patchUrls = @(
           "https://catalog.s.download.windowsupdate.com/d/msdownload/update/software/secu/2024/11/windows10.0-kb5046615-x64_d88444ff882129230297936cb0a221157099d074.msu",
-          "https://catalog.s.download.windowsupdate.com/d/msdownload/update/software/secu/2024/09/windows10.0-kb5044016-x64_64b2bcbd9edcc7fc1dd860751fdb537c947cf037.msu"
+          "https://catalog.s.download.windowsupdate.com/d/msdownload/update/software/secu/2024/09/windows10.0-kb5044016-x64_64b2bcbd9edcc7fc1dd860751fdb537c947cf037.msu",
+          "https://catalog.s.download.windowsupdate.com/d/msdownload/update/software/secu/2024/12/windows10.0-kb5048661-x64_6bfc43258aab46ec793d2977122226a4a2b8ed6f.msu"
         )
-
         $global:patchIDs = @(
           "KB5046615",
```
Comment (AbelHu) [inline vhdbuilder/packer/generate-windows-vhd-configuration.ps1:71]: Remove KB5046615?

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 [hunk 2]
```diff
@@ -60,16 +60,17 @@ $global:defenderUpdateInfoUrl = "https://go.microsoft.com/fwlink/?linkid=870379&
...
-          "KB5044016"
+          "KB5044016",
```
Comment (AbelHu) [inline vhdbuilder/packer/generate-windows-vhd-configuration.ps1:72]: Update to KB5046540? https://support.microsoft.com/en-us/topic/november-12-2024-kb5046540-cumulative-update-for-net-framework-3-5-4-7-2-and-4-8-for-windows-10-version-1809-and-windows-server-2019-308fc49b-cb66-45d1-a78f-a9050e07ea86

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 [hunk 3]
```diff
@@ -147,17 +148,14 @@ $global:imagesToPull += @(
     # OMS-Agent (Azure monitor). Owner: ganga1980 (Ganga Mahesh Siddem)
     "mcr.microsoft.com/azuremonitor/containerinsights/ciprod:win-3.1.24",
 
-    # CNS (Container Networking Service) Owner: jaer-tsun (Jaeryn)
-    "mcr.microsoft.com/containernetworking/azure-cns:v1.4.52",
```
Comment (AbelHu) [inline vhdbuilder/packer/generate-windows-vhd-configuration.ps1:151]: v1.4.52 is still used.

Comment (AbelHu) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:151]: v1.4.56 is not used

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 [hunk 4]
```diff
@@ -60,16 +60,17 @@ $global:defenderUpdateInfoUrl = "https://go.microsoft.com/fwlink/?linkid=870379&
 
 switch -Regex ($windowsSku) {
     "2019-containerd" {
-        # TODO: clean up when you bump 12B. These patches do an upgrade from 10B to 12B in a way that doesn't break openssh.
```
Comment (AbelHu) [inline vhdbuilder/packer/generate-windows-vhd-configuration.ps1:63]: Does you still hit the openssh permissions problem with WS2019 base image with 12b?

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 (hunk 3 above)
Comment (timmy-wright) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:151]: Changed.

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 (hunk 4 above)
Comment (timmy-wright) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:63]: I sure did! Was super annoying.

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 (hunk 1 above)
Comment (timmy-wright) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:72]: I think it's still needed due to the openssh issue with win2019.

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

**Pull request merged**